File: leetcode-everyday/双指针/680-验证回文字符串II/680-验证回文字符串II.js
/**
 * @param {string} s
 * @return {boolean}
 */
 var validPalindrome = function(s) {
  let left = 0;
  let right = s.length-1;    
  while(left <= right) {
      if(s.charAt(left) === s.charAt(right)) {
          left ++;
          right --;
      } else {
          return helper(s, left+1, right) || helper(s, left, right-1);
      }
  }
  return true;
};

function helper(s, left, right) {
  while(left < right) {
      if(s.charAt(left) === s.charAt(right)) {
          left ++;
          right --;
      } else {
          return false;
      }
  }
  return true;
}

File: leetcode-everyday/双指针/680-验证回文字符串II/680-验证回文字符串II.cpp
class Solution {
private:
    bool helper(string s, int left, int right) {
        while(left < right) {
            if(s[left] == s[right]) {
                left ++, right--;
            } else {
                return false;
            }
        }   
        return true;     
    }
public:
    bool validPalindrome(string s) {
        int left = 0, right = s.size()-1;
        while(left < right) {
            if(s[left] == s[right]) {
                left++, right--;
            } else {
                return helper(s, left+1, right) || helper(s, left, right-1);
            }
        }
        return true;
    }    
};